distributed/tests/test_client.py

```diff
@@ -7345,6 +7334,31 @@ async def test_dump_cluster_state_json(c, s, a, b, tmp_path, local):
         await c.dump_cluster_state(filename, format="json")
 
 
+@pytest.mark.parametrize("local", [True, False])
+@pytest.mark.parametrize("_format", ["msgpack", "yaml"])
+@pytest.mark.parametrize("workers", [True, False])
+@gen_cluster(client=True)
+async def test_inspect_cluster_dump(c, s, a, b, tmp_path, _format, local, workers):
+    filename = tmp_path / "foo"
+    if not local:
+        pytest.importorskip("fsspec")
+        # Make it look like an fsspec path
+        filename = f"file://{filename}"
+
+    A = da.ones(100, chunks=25)
+    await c.persist(A)
+    await c.dump_cluster_state(filename, format=_format)
+
+    suffix = ".gz" if _format == "msgpack" else ""
+    inspector = DumpInspector(f"{filename}.{_format}{suffix}")
+    tasks = inspector.tasks_in_state("memory", workers=workers)
+    assert set(tasks.keys()) == set(map(str, A.__dask_keys__()))
+    it = iter(tasks.keys())
+    stories = inspector.story(next(it), workers=workers)
```

Comment:
I'm wondering if this output works with `distributed.utils_test.assert_worker_story` or if this is another format

Reply:
This is a good question. I do not think it will work with `assert_worker_story` as `DumpInspector.story` returns a story concatenated from stories on both scheduler and workers. These formats are not the same, but I wonder if this matters in the context of debugging? This also relates to your question regarding mixing scheduler and worker tasks in https://github.com/dask/distributed/pull/5920#discussion_r825779788.

Reply:
There are now separate worker and scheduler story functions. `assert_worker_story` is used to test the worker story.